### ISSUE TITLE
Using -g retail not using the Interface-Retail value in TOC

### ISSUE THREAD

**Issue by CDAGaming** [BigWigsMods/packager#87]
My mod's TOC uses the following values:
```
## Interface: 90100
## Interface-Retail: 90005
## Interface-Classic: 11307
## Interface-BC: 20501
```
Yet in the following action, it uses the main Interface value: https://github.com/CDAGaming/CraftPresence-Wow-Edition/runs/2402189280?check_suite_focus=true

The preferred and intended behavior here is that the Interface-Retail value should take priority over the Interface value, especially if it is explicitly stated that it's being built for retail.

**Comment by CDAGaming** [comment]
After some debugging, this line appears to be the root cause of the issue:

`if [[ -z "$toc_version" ]] || [[ "$game_type" == "classic" && "$toc_version" != "113"* ]] || [[ "$game_type" == "bc" && "$toc_version" != "205"* ]] || [[ "$game_type" == "retail" && ("$toc_version" == "113"* || "$toc_version" == "205"*) ]]; then`

The issue with this line is that it doesn't take into account the scenario that both an Interface line (root_toc_version) and the -Retail line can both exist, and is instead relying that the root_toc_version will always be at Retail Value (Or in other words higher then any other flavor.

A way this could be fixed is by checking aditionally if an `Interface-flavorHere` line exists that matches the specified flavor, but this can easily conflict with the `@` arguments. Hence the method I'm going to try out is by making it so that if you specify in the args a specific flavor, an additional variable is in the script stating that you were specific, hence using the check.

**Comment by LoneWanderer-GH** [comment]
I've been testing current release and also the proposed merge request.
Both seem to not work for me.

TOC file:
```
#@debug@
# for personnal dev purposes
## Interface: 11307
#@end-debug@
## Interface-Retail: 90005
## Interface-Classic: 11307

# other stuff
# ...
```

The following command lines result in no `## Interface: ...` **at all** in produced TOC file.
```
./release.sh -z -e -d -g classic
./release.sh -z -e -d -g retail
```
Am I missing something ?

-----
AFAIU Test cases would be, with the following input TOC file:

```
## Interface: 11307
## Interface-Retail: 90005
## Interface-Classic: 11307
```

`./release.sh -z -e -d -g classic` resulting in:
```
## Interface: 11307
```


`./release.sh -z -e -d -g retail` resulting in:
```
## Interface: 90005
```

`./release.sh -z -e -d -g 1.23.45` resulting in:
```
## Interface: 1.23.45
```
(feel free to add Burning Crusade stuff  to these)

**Comment by nebularg** [comment]
> ```
> #@debug@
> # for personnal dev purposes
> ## Interface: 11307
> #@end-debug@
> ## Interface-Retail: 90005
> ## Interface-Classic: 11307
> 
> # other stuff
> # ...
> ```

Why are you using debug tags instead of just leaving the Interface value 11307?

**Comment by LoneWanderer-GH** [comment]
Debug tags removal are not really important there.
Problem is fairly similar to yours : as I wrote already, I have 0 resulting `## Interface:` tag in file.

I (wrongly ?) expected the retail/classic/bc to take precedence.

**Comment by nebularg** [comment]
It is important because without them it'll work as expected. They serve no purpose as is

**Comment by CDAGaming** [comment]
Going to adjust the merge request temporarily to have additional logging to show the value of the defined field. Please retest after that commit comes up and send me the output.

**Comment by nebularg** [comment]
@CDAGaming don't bother, I'll fix it so interface is used as is when no version is set

**Comment by nebularg** [comment]
Fixed in 1701d5e and 1701d5e

**Comment by LoneWanderer-GH** [comment]
Used it in several GitHub workflows, works like a charm !👍🏻
